Common/src/main/java/com/blamejared/crafttweaker/api/ScriptLoadingOptions.java

```diff
@@ -111,8 +120,51 @@ public ScriptLoadingOptions setSource(ScriptLoadSource source) {
         return this;
     }
     
-    public static record ScriptLoadSource(ResourceLocation id) {
+    public ScriptLoadingOptions setRunConfiguration(final ScriptRunConfiguration runConfiguration) {
+        
+        this.runConfiguration = LegacyRunConfig.adapt(runConfiguration);
+        return this;
+    }
+    
+    @Deprecated(forRemoval = true) // Replace with equivalent of LegacyRunConfig#adapt
+    public void configureRun(final IgnorePrefixCasingBracketParser bep, final ScriptingEngine engine) throws CompileException {
+        
+        this.runConfiguration.configureRun(bep, engine);
+    }
+    
+    @Deprecated(forRemoval = true)
+    private void legacyConfig(final IgnorePrefixCasingBracketParser bep, final ScriptingEngine engine) throws CompileException {
```

Comment:
So, we create a run config that whenever called creates another run config and executes it?
If the log message and the old event are the only reason for that, why not create a child class of the default config that does these two before calling `super...`

Edit: But for that you would need to implement the default mechanism as class and not as method reference.